app/src/main/java/subins2000/manglish/MainActivity.java

```diff
@@ -1,75 +1,96 @@
  */
 
 package subins2000.manglish;
 
+import android.content.SharedPreferences;
 import android.net.Uri;
 import android.os.Bundle;
+
 import com.google.android.material.navigation.NavigationView;
 
+import android.preference.PreferenceManager;
 import android.view.Menu;
 import android.view.MenuItem;
+import android.view.View;
+import android.view.inputmethod.InputMethodManager;
 
+import androidx.annotation.NonNull;
 import androidx.appcompat.app.ActionBarDrawerToggle;
 import androidx.appcompat.app.AppCompatActivity;
 import androidx.appcompat.widget.Toolbar;
 import androidx.core.view.GravityCompat;
 import androidx.drawerlayout.widget.DrawerLayout;
 import androidx.fragment.app.Fragment;
-import androidx.fragment.app.FragmentTransaction;
 
 public class MainActivity extends AppCompatActivity
-        implements NavigationView.OnNavigationItemSelectedListener, HomeFragment.OnFragmentInteractionListener, AboutFragment.OnFragmentInteractionListener {
+        implements NavigationView.OnNavigationItemSelectedListener,
+        HomeFragment.OnFragmentInteractionListener, AboutFragment.OnFragmentInteractionListener {
+
+    private static final String NAV_ID = "NavId";
 
+    DrawerLayout drawer;
     NavigationView navigationView;
 
-    Fragment homeFragment = new HomeFragment();
-    Fragment aboutFragment = new AboutFragment();
+    SharedPreferences prefs;
+
+    int activeNavId = R.id.nav_home;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        prefs = PreferenceManager.getDefaultSharedPreferences(this);
+
         setContentView(R.layout.activity_main);
-        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        Toolbar toolbar = findViewById(R.id.toolbar);
         setSupportActionBar(toolbar);
 
-        DrawerLayout drawer = (DrawerLayout) findViewById(R.id.drawer_layout);
+        drawer = findViewById(R.id.drawer_layout);
         ActionBarDrawerToggle toggle = new ActionBarDrawerToggle(
                 this, drawer, toolbar, R.string.navigation_drawer_open, R.string.navigation_drawer_close);
         drawer.addDrawerListener(toggle);
+        drawer.addDrawerListener(new DrawerLayout.SimpleDrawerListener() {
+            @Override
+            public void onDrawerSlide(View drawerView, float slideOffset) {
+                if(slideOffset > 0) tryHideKeyboard();
+            }
+        });
         toggle.syncState();
 
         navigationView = findViewById(R.id.nav_view);
         navigationView.setNavigationItemSelectedListener(this);
-
         navigationView.getMenu().getItem(0).setChecked(true);
 
-        // Add fragments
-        FragmentTransaction transaction = getSupportFragmentManager().beginTransaction();
-        transaction.add(R.id.frame, homeFragment);
-        transaction.add(R.id.frame, aboutFragment);
-        transaction.commit();
+        if (savedInstanceState == null) {
+            setFragment(new HomeFragment());
```

Comment:
Instead pass the saved object ?

Reply:
In this case `FragmentManager` stores last fragment in the container with its state by itself. `Fragment` lifecycle is attached to the activity lifecycle. No need to manually add fragment after recreation if  `savedInstanceState != null`. It causes fragment overlay (new fragment adds over the old). We can test my code with debugger and make sure it works better ;)